WhoIsThatServer.Recognition/Controllers/RecognitionServicesController.cs

```diff
@@ -31,10 +31,15 @@ public async Task<JsonResult> InitiateRecognition()
                 return Json(RecognitionErrorMessages.NoFacesFoundError);
             }
 
-            catch (ArgumentNullException argumentNullException)
+            catch (ArgumentNullException imageArgumentNullException) when (imageArgumentNullException.Message.Contains("image"))
```

Comment:
Another class and == would be much cleaner but that may do for now

Reply:
Alright, anyhow, will change it to your suggestion with manager exception, it looks very good